### DISCUSSION TITLE
External access to dependent container

### DISCUSSION THREAD

**Discussion by sam-semillia** [bubuntux/nordlynx#105] [Q&A] [answered]
Hi,

I want to run NordLynx VPN container, together with another container (OSCAM) to tunnel all OSCAM traffic through NordVpn.
OSCAM should also be availaible from external access as well. Unfortunately external access only works sometimes, most of the time it does not. Internal connections work perfectly fine. The port is forwarded in the router. This is running on docker ubuntu x64.

Can someone help me please?

Thanks, BR 

version: "3"
services:
  vpn:
    container_name: NordVPN_NordLynx
    image: ghcr.io/bubuntux/nordlynx
    network_mode: bridge
    ports:
      - 12357:12357/udp #foward to OSCAM
    cap_add:
      - NET_ADMIN #required
    environment:
     - PUID=1000
     - PGID=1000
     - PRIVATE_KEY=
     - NET_LOCAL=192.168.1.0/24
     - ALLOWED_IPS=0.0.0.0/0
     - TZ=Europe/London
    sysctls:
     - net.ipv6.conf.all.disable_ipv6=1  # Recomended if using ipv4 only      
     - net.ipv4.conf.all.src_valid_mark=1
    restart: unless-stopped
  oscam:
    image: lscr.io/linuxserver/oscam:latest
    container_name: Oscam_VPN
    network_mode: service:vpn
    environment:
      - PUID=0
      - PGID=0
      - TZ=Europe/London
    volumes:
      - /home/oscam:/config
    devices:
      - /dev/ttyUSB0:/dev/ttyUSB0
    privileged: true
    restart: unless-stopped
    depends_on:
      - vpn

**Comment by sam-semillia** [comment]
No ideas?
Thanks, BR

**Comment by Gunther-Schulz** [comment]
This has been frustrating for me too. From host network no problem, from external ip no dice.

**Reply by Gunther-Schulz** [reply]
Well, I got it figured out. You need to add a forwarding proxy that is not in the same network namespace as the VPN to your docker-compose file and you set it to forward to the app that runs in the VPN namespace.

**Reply by Legomaniac** [reply]
@Gunther-Schulz I know this post is old, but I'm trying to accomplish the exact same thing and having a hell of a time.

Assuming you're still using this setup, could you share your docker-compose? If not, could you share what you're using now? Thanks!

**Reply by sam-semillia** [reply]
Hey, anyone could share his working docker compose please?

**Comment by sam-semillia** [comment, marked as answer]
found a way by running an additional nginx, not running through vpn container and passing on the necessary ports
